patch/0005-Support-Centec-TsingMa-SOC-MTD.patch

```diff
@@ -0,0 +1,406 @@
++	sdhci_writel(host, avg, REG_OFFSET_ADDR + AT_STAT_R);
++
++	dev_info(mmc_dev(host->mmc), "Tuning %s at 0x%x ret %d\n",
++		 ret ? "failed" : "passed", avg, ret);
++
++	return ret;
++}
++
++static const struct sdhci_ops sdhci_ctc5236_ops = {
++	.read_w = sdhci_ctc5236_readw,
++	.read_l = sdhci_ctc5236_readl,
++	.set_clock = ctc_sdhci_set_clock,
++	.set_bus_width = sdhci_set_bus_width,
++	.reset = sdhci_ctc5236_reset,
++	.set_uhs_signaling = sdhci_set_uhs_signaling,
++	.get_max_clock = sdhci_pltfm_clk_get_max_clock,
++	.platform_execute_tuning = sdhci_ctc5236_execute_tuning,
++};
++
++static struct sdhci_pltfm_data sdhci_ctc5236_pdata = {
++	.ops = &sdhci_ctc5236_ops,
++	.quirks2 = SDHCI_QUIRK2_PRESET_VALUE_BROKEN | SDHCI_QUIRK2_BROKEN_HS200,
++	.quirks = SDHCI_QUIRK_CAP_CLOCK_BASE_BROKEN | SDHCI_QUIRK_BROKEN_ADMA,
++};
++
++static int sdhci_ctc5236_probe(struct platform_device *pdev)
++{
++	struct sdhci_host *host;
++	struct sdhci_pltfm_host *pltfm_host;
++	struct clk *clk;
++	int ret;
++
++	host = sdhci_pltfm_init(pdev, &sdhci_ctc5236_pdata, 0);
++	if (IS_ERR(host))
++		return PTR_ERR(host);
++
++	clk = devm_clk_get(&pdev->dev, "mmc_clk");
++	if (IS_ERR(clk)) {
++		dev_err(&pdev->dev, "Peripheral clk not found\n");
++		return PTR_ERR(clk);
++	}
++	pltfm_host = sdhci_priv(host);
++	pltfm_host->clk = clk;
++	clk_prepare_enable(clk);
++
++	regmap_base =
++	    syscon_regmap_lookup_by_phandle(pdev->dev.of_node, "ctc,sysctrl");
++	if (IS_ERR(regmap_base))
++		return PTR_ERR(regmap_base);
++
++	mmc_of_parse_voltage(pdev->dev.of_node, &host->ocr_mask);
++
++	ret = mmc_of_parse(host->mmc);
++	if (ret)
++		goto err_sdhci_add;
++
++	ret = sdhci_add_host(host);
++	if (ret)
++		goto err_sdhci_add;
++	return 0;
++
++err_sdhci_add:
++	sdhci_pltfm_free(pdev);
++	return ret;
++}
++
++static const struct of_device_id sdhci_ctc5236_of_match[] = {
++	{.compatible = "centec,ctc5236-sdhci"},
++	{},
++};
++
++MODULE_DEVICE_TABLE(of, sdhci_ctc5236_of_match);
++
++static struct platform_driver sdhci_ctc5236_driver = {
++	.driver = {
++		   .name = "sdhci-ctc5236",
++		   .pm = &sdhci_pltfm_pmops,
++		   .of_match_table = of_match_ptr(sdhci_ctc5236_of_match),
++		   },
++	.probe = sdhci_ctc5236_probe,
++	.remove = sdhci_pltfm_unregister,
++};
++
++module_platform_driver(sdhci_ctc5236_driver);
++
++MODULE_DESCRIPTION("SDHCI driver for Centec TsingMa SoCs");
++MODULE_AUTHOR("Jay Cao <caoj@centecnetworks.com>");
++MODULE_LICENSE("GPL v2");
+diff --git a/drivers/mtd/devices/m25p80.c b/drivers/mtd/devices/m25p80.c
+index 270d3c9..76b3aa9 100644
+--- a/drivers/mtd/devices/m25p80.c
++++ b/drivers/mtd/devices/m25p80.c
+@@ -299,7 +299,7 @@ static const struct spi_device_id m25p_ids[] = {
+ 	{"m25p40"},	{"m25p80"},	{"m25p16"},	{"m25p32"},
+ 	{"m25p64"},	{"m25p128"},
+ 	{"w25x80"},	{"w25x32"},	{"w25q32"},	{"w25q32dw"},
+-	{"w25q80bl"},	{"w25q128"},	{"w25q256"},
++	{"w25q80bl"},	{"w25q128"},	{"w25q256"},	{"mx25u3235f"},
```

Comment:
why is this needed? 

starting from v5.4, this file has been removed.

https://github.com/torvalds/linux/commit/b35b9a10362d203451d920d01ab8d6cd55cbaf2a